### DISCUSSION TITLE
New Relic Integration

### DISCUSSION THREAD

**Discussion by sonicaghi** [Kong/kong#7365] [Ideas and feature requests]

**Comment by catchdave** [comment]
+1

**Comment by vandiedakaf** [comment]
+1

**Comment by pavkatar** [comment]
+1

**Comment by m1k3** [comment]
+1

**Comment by jcoulehan** [comment]
+1

**Comment by faabiosr** [comment]
+1

What is the level of integration with New Relic you want?

**Comment by d4v3y0rk** [comment]
+1

**Comment by joseandrespg** [comment]
+1

**Comment by JnMik** [comment]
+1

**Comment by sebastianhutter** [comment]
+1

**Comment by tiny-dancer** [comment]
+1 yay

**Comment by coopr** [comment]
Tell me more - what would you want this to do? With which parts of New Relic would you like to see Kong integrated? (Plugins, Insights, Infrastructure - maybe APM?)

What sort of monitoring are you already doing, and how is it not serving your needs? 

(FYI I'm Product Management here at Mashape - hence all my questions - thanks in advance for your answers!)

**Comment by jcoulehan** [comment]
I guess the most appropriate place would be as a plugin. We have started using a Cassandra plugin to monitor Kong's Cassandra DB.

We have just started using Kong and are making it an important part of our software infrastructure.  At the moment we are not actually monitoring Kong at all, just the availability of the back-end services. 

I think at a bare minimum a Kong up/down count so that we can generate alerts from NR when the Kong service is down. Also maybe some basic request numbers and response times.

If there is a better way to doing this, or another way please let me know.

Thanks for asking.

**Comment by rcyrus** [comment]
@coopr honestly, all of those things would be great. We make heavy use of nearly all of new relic features. we keep tabs on infrastructure as well as use insights for usage reporting. It would be nice to be able to keep tabs on kong node health as well as track usage of various APIs and any other metrics we could drill into.

**Comment by fredericocabral** [comment]
+1

**Comment by hutchic** [comment]
I agree @jcoulehan a plugin would be the ideal place for this functionality.

A good spot to grab prior art from the sounds of what you'd like to capture would be the [DataDog](https://github.com/Mashape/kong/tree/master/kong/plugins/datadog) [plugin](https://getkong.org/plugins/datadog/) which currently captures:
```
request_count
request_size
response_size
latency
server
status_count
unique_users
request_per_user
```

The [plugin development](https://getkong.org/docs/0.10.x/plugin-development/) guide and [kong vagrant](https://github.com/Mashape/kong-vagrant#developing-plugins) is a good launching point if you wanted to take an initial stab at it

**Comment by JnMik** [comment]
I've built this for my needs, might help some of you ? 
https://github.com/JnMik/kong-plugin-newrelic-insights

Cheers

**Comment by kigsmtua** [comment]
+1

**Comment by mo-mughrabi** [comment]
+1

**Comment by PFadel** [comment]
any news about this ?

**Comment by rbabyuk-vs** [comment]
I am interested in this one as well. +1

**Comment by premvilas30** [comment]
do we have any update on this, is this in pipeline or something?
